variant_annotator.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python3
+
+import csv
+import requests
+import time
+import sys
+
+
+class VariantAnnotator:
+    
+    variant_list = []
+    
+    def __init__(self) -> None:
+        timestr = time.strftime("%Y%m%d-%H%M%S")
+        self.filename_out = 'annotated_data_' + timestr + '.tsv'
+        self.create_output_file()
+        self.read_stdin_clean()
+    
+    def read_stdin_clean(self):
+        # TODO: check for stdin
+        for line in sys.stdin:
+            self.variant_list.append(line.strip())  # Remove trailing whitespace and return chars
+            
+    def get_annotations(self, variant_in) -> dict:    
+        url = 'https://rest.ensembl.org/vep/human/hgvs/' + variant_in
+        response = requests.get(url, headers={"Content-Type" : "application/json"})
+        
+        # Only process successful (200) responses
+        if response.status_code == 200:
+            annotation_json_data = response.json()[0]  # json object is returned in a one item list. just return the zero item for ease of reading
+
+            # Put data in a dict for later processing. Dict is good as we can reference by name later.
+            # TODO: check for missing data? Not sure.
+            annotation_dict = {
+                'variant': variant_in,
+                'assembly_name': annotation_json_data['assembly_name'],
+                'seq_region_name': annotation_json_data['seq_region_name'],
+                'start': annotation_json_data['start'],
+                'end': annotation_json_data['end'],
+                'most_severe_consequence': annotation_json_data['most_severe_consequence'],
+                'strand': annotation_json_data['strand'],
+                'gene_symbol': annotation_json_data['transcript_consequences'][0]['gene_symbol']
+            }
+            # print(variant_in, file=sys.stdout)
+            return annotation_dict
+        else:
+            print('The input value: "' + variant_in + '" failed with a response of: "' + str(response.status_code) + '"', file=sys.stderr)
+            return {} # This could also return None but we returns a dict for funtion output consistancy
+            
+    def create_output_file(self)-> None:
+        # TODO: fix the path that this file gets written to
+        with open(self.filename_out, 'w', newline='') as tsvfile:
+            writer = csv.writer(tsvfile, delimiter='\t', lineterminator='\n')
+            writer.writerow(['variant', 'assembly_name', 'seq_region_name', 'start', 'end', 'most_severe_consequence', 'strand', 'genes'])
+            
+
+    def process_results_to_file(self, annotation_in) -> None:
```

Comment:
It makes sense to me to isolate the processing of results from the writing to a file. That way, if we want to use the results in a different flow we aren't bound to them being written to a file. This function could take in an annotation and return the standardized version, then the create_output_file could accept a list of the standardized entries and write all of them to a file together